src/v3/src/transformer/phone/phoneEnrollmentTransformer.ts

```diff
@@ -76,37 +77,62 @@ export const transformPhoneEnrollment: IdxStepTransformer = ({ formBag, transact
     },
   };
 
-  const stepper: StepperLayout = {
-    type: UISchemaLayoutType.STEPPER,
-    elements: [
-      {
-        type: UISchemaLayoutType.VERTICAL,
-        elements: [
-          smsInfoTextElement,
-          methodTypeElement,
-          phoneNumberElement,
-          smsStepSubmitButton,
-        ],
-      },
-      {
-        type: UISchemaLayoutType.VERTICAL,
-        elements: [
-          voiceInfoTextElement,
-          methodTypeElement,
-          phoneNumberElement,
-          voiceStepSubmitButton,
-        ],
-      },
-    ],
+  const phoneOptionLayouts: { [key:string]: UISchemaLayout | StepperLayout } = {
+    sms: {
+      type: UISchemaLayoutType.VERTICAL,
+      elements: [
+        smsInfoTextElement,
+        phoneNumberElement,
+        smsStepSubmitButton,
+      ],
+    },
+    voice: {
+      type: UISchemaLayoutType.VERTICAL,
+      elements: [
+        voiceInfoTextElement,
+        phoneNumberElement,
+        voiceStepSubmitButton,
+      ],
+    },
+    both: {
+      type: UISchemaLayoutType.STEPPER,
+      elements: [
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            smsInfoTextElement,
+            methodTypeElement,
+            phoneNumberElement,
+            smsStepSubmitButton,
+          ],
+        },
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            voiceInfoTextElement,
+            methodTypeElement,
+            phoneNumberElement,
+            voiceStepSubmitButton,
+          ],
+        },
+      ],
+    } as StepperLayout,
   };
 
   uischema.elements = [
     titleElement,
-    stepper,
   ];
 
+  const phoneMethodOptions = methodTypeElement!.options!.inputMeta.options!;
+  if (phoneMethodOptions.length === 1) {
+    const methodType = phoneMethodOptions[0].value.toString();
+    uischema.elements.push(phoneOptionLayouts[methodType]);
+  } else {
+    uischema.elements.push(phoneOptionLayouts.both);
```

Comment:
In this condition, you can move line 42 here:
`data['authenticator.methodType'] = methodTypeElement!.options!.inputMeta.options![0].value;`

Reply:
It looks like this line needs to before outside before the condition, or else it wont show the extension input in the case where voice is the only option

Reply:
Ohhh that is right, the phone authenticator needs that value for the extension 

Ok, I'm sorry, scratch that and the button action params I discussed above. I think we should be safe by just setting the value directly in data (I dont think it will be overwritten).

Reply:
okay got it no prob!